tests: net: Run networking tests only for selected platforms

jukkar opened the pull request:
Run networking tests for native_posix, qemu_x86 and qemu_cortex_m3
platforms only as the tests are generic enough so no need to run
them in real physical device.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar committed:
tests: net: Run networking tests only for selected platforms

Run networking tests for native_posix, qemu_x86 and qemu_cortex_m3
platforms only as the tests are generic enough so no need to run
them in real physical device.

Fixes #9623 #9614 #9622 #9621 #9618

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
We could even lower the number of tests if we skip testing in qemu_x86, that would lower the test count from 147 to 103 for tests/net directory. If this would be ok, I can send a new version that does that.

jukkar commented:
Added fixed bugs list in commit message.

pfalcon commented:
> We could even lower the number of tests if we skip testing in qemu_x86

I think that would go too far.

pfalcon commented:
Trying this patch together with https://github.com/zephyrproject-rtos/zephyr/pull/7831 (in the hope that it'll be possible to merge it afterwards!), and it uncovers some missed samples: samples/net/telnet, samples/net/dhcpv4_client, samples/net/dns_resolve, samples/net/nats, samples/net/syslog_net

jukkar commented:
This PR only checks things under tests/net directory which was the whole point of this PR. It probably does not make sense to try to prevent sanitychecker from using samples/net programs.

pfalcon commented:
> This PR only checks things under tests/net directory which was the whole point of this PR. It probably does not make sense to try to prevent sanitychecker from using samples/net programs.

So, I don't see any good points in this PR - it decreases *build* coverage, which is generally bad, but could work as a stopgap measure to address https://github.com/zephyrproject-rtos/zephyr/issues/9015 until better solution comes up. But it stops short of fully masking https://github.com/zephyrproject-rtos/zephyr/issues/9015 and being able to resolve https://github.com/zephyrproject-rtos/zephyr/pull/7831.

jukkar commented:
This PR does not decrease build coverage, it merely prevents unnecessary builds for tests/net directory. Those unit tests can safely be run in couple of architectures only.

pfalcon commented:
> This PR does not decrease build coverage

What about network driver code coverage from platforms which get excluded?


Anyway, I guess the best thing for me is to approve this, even if this doesn't clear the way for #7831, it goes in the needed direction (even if needed direction is a stopgap measure, again). (Otherwise, I thought someone else approves it anyway, surprised nobody did yet.)

jukkar committed:
tests: net: websocket: Fix crash when tearing down tests

No need to have setup and teardown functions. Teardown will
even cause crash when tearing down last IPv4 test.

Fixes #9617

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

nashif commented:
well, tbh I do not like the approach here, we need to do this better, but right now this is the only way I can see it done. I will try to streamline how this will be done in sanitycheck in testing in general after 1.13.
